overlord/devicestate/handlers_systems.go

```diff
@@ -262,6 +286,34 @@ func (m *DeviceManager) doCreateRecoverySystem(t *state.Task, _ *tomb.Tomb) (err
 	if err := setTaskRecoverySystemSetup(t, setup); err != nil {
 		return fmt.Errorf("cannot record recovery system setup state: %v", err)
 	}
+
+	// if the caller did not request to test the system, then we immediately
+	// promote the system and mark it as ready to use
+	if !setup.TestSystem {
+		if err := boot.PromoteTriedRecoverySystem(remodelCtx, label, []string{label}); err != nil {
+			return fmt.Errorf("cannot promote recovery system %q: %v", label, err)
+		}
+
+		if setup.MarkCurrent {
+			if err := m.recordSeededSystem(st, &seededSystem{
+				System:    label,
+				Model:     model.Model(),
+				BrandID:   model.BrandID(),
+				Revision:  model.Revision(),
+				Timestamp: model.Timestamp(),
+				SeedTime:  time.Now(),
+			}); err != nil {
+				return fmt.Errorf("cannot record a new seeded system: %v", err)
+			}
+		}
+
+		if err := boot.MarkRecoveryCapableSystem(label); err != nil {
+			return fmt.Errorf("cannot mark system %q as recovery capable", label)
+		}
```

Comment:
some of this and repeated similar code below (or the undo equivalent) should be organized in a couple small helpers?

Reply:
Good thought, thanks.